nmrfx-core/src/main/java/org/nmrfx/chemistry/io/PDBFile.java

```diff
@@ -1041,15 +1045,15 @@ public static Compound readResidue(String fileName, String fileContent, Molecule
             } catch (IOException ioe) {
                 System.err.println(ioe.getMessage());
                 molecule.getAtomTypes();
-                if (calcBonds) {
+                if (calcBonds && compound != null) {
                     compound.calcAllBonds();
                 }
                 return compound;
```

Comment:
It seems a bit strange to return a compound that may not have had everything added to it from the file, it was specifically changed to do this in 2018 though so left it like this...

Reply:
This is OK with your added test